### PR TITLE
Implementation of class Item

### LINKED ISSUE #1: [1pt] Create Item class
Create an Item class in a separate .rb file

### LINKED ISSUE #2: [1pt] Class Items: Add all methods visible in the diagram.
Add all methods for the item class visible in the diagram 

Image

### LINKED ISSUE #3: [1pt] Implement methods: can_be_archived?() in the Item class should return true if published_date is older than 10 years
Define can_be_archived_class? () in time . If present  & year are bigger than 10 years from published_date.

### LINKED ISSUE #4: [1pt] Implement methods: can_be_archived?() in the Item class should return true. otherwise, it should return false
Define can_be_archived_class? () in time . If present & year are bigger than 10 years from published_date became false.

### LINKED ISSUE #5: [1pts] move_to_archive() in the Item class should reuse can_be_archived?() method

### LINKED ISSUE #6: [1pt] Item class: move_to_archive() when can_be_archived? is true
**move_to_archive()** should change the archived property if the result of the **can_be_archived?()** method is true and save the object to the archive.json file.

### LINKED ISSUE #7: move_to_archive() in the Item class. should do nothing if the result of the can_be_archived?() method is false

### PR DESCRIPTION
**Pull request by carlosmuerto**
# Implementation of class Item
- close #1 
- close #2
- close #3
- close #4
- close #5
- close #6
- close #7
- add unit test for item

### CODE REVIEW

File: spec/item_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+describe Item do
+  before :each do
+    @archevable_item = Item.new(
+      'genre',
+      'author',
+      'source',
+      'label',
+      Time.now - (60 * 60 * 24 * 365 * 15)
+    )
+    @non_archevable_item = Item.new(
```
Comment (andym80) [inline spec/item_spec.rb:13]: archivable

File: spec/item_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      'genre',
+      'author',
+      'source',
+      'label',
+      Time.now - (60 * 60 * 24 * 365 * 5)
+    )
+  end
+
+  context 'Item Id is auto_incremente' do
+    it 'non_archevable_item and archevable_item has diferent ids' do
```
Comment (andym80) [inline spec/item_spec.rb:23]: archivable

**Review by andym80 — COMMENTED** [review]
Great implementation at  **item.rb** file. 
Please just correct some words marked to get more understood in the code.

**Review by andym80 — CHANGES_REQUESTED** [review]
Great implementation at  **item.rb** file. 
Please just correct some words marked to get more understood in the code.

File: spec/item_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      expect(@archevable_item.id != @non_archevable_item).to be(true)
+    end
+  end
+
+  context 'Item Id is auto_incremente' do
+    it 'archevable_item move_to_archive DOES archive the file' do
+      expect(@archevable_item.archived).to be(false)
+      @archevable_item.move_to_archive
```
Comment (andym80) [inline spec/item_spec.rb]: archivable

File: spec/item_spec.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      expect(@archevable_item.archived).to be(true)
```
Comment (andym80) [inline spec/item_spec.rb]: archivable

File: spec/item_spec.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    end
+
+    it 'non_archevable_item move_to_archive DOES NOT archive the file' do
```
Comment (andym80) [inline spec/item_spec.rb]: archivable

File: spec/item_spec.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+      expect(@non_archevable_item.archived).to be(false)
```
Comment (andym80) [inline spec/item_spec.rb]: archivable

File: spec/item_spec.rb [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+      @non_archevable_item.move_to_archive
+      expect(@non_archevable_item.archived).to be(false)
```
Comment (andym80) [inline spec/item_spec.rb]: archivable 
you get it the last one !

**Review by AmaliaGM — CHANGES_REQUESTED** [review]
#62

**Review by andym80 — APPROVED** [review]

**Review by AmaliaGM — APPROVED** [review]

**Pull request merged**